src/nvim/eval.c

```diff
@@ -16682,16 +16631,16 @@ static void f_termopen(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   (void)setfname(curbuf, (char_u *)buf, NULL, true);
```

Comment:
In the old code `terminal_open()` was only called after the term title was created and saved.
Now `channel_terminal_open()` (further up) calls `terminal_open()` before the term title is ready.